Setup doc updates

pzrq opened the pull request:
Drop https://gitter.im from all setups, update OSX setup to minimal `nvm` setup needed to get Compass running.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mongodb-js/mongodb-js/3)

<!-- Reviewable:end -->

pzrq committed:
OSX: Optimistically drop easy_install, pip, pyobjc; add nvm

Looks like nothing is using it. We can add it back if something in JS is somehow using it, but more likely we should work to drop it as a dependency.

pzrq committed:
Drop gitter

Slack > gitter, but that's in the communication doc for now.

pzrq committed:
OSX: Optimistically drop node-gyp global dependency

It should be documented in each individual repo if it is actually needed, and preferably not as a global. Further it's not part of Windows/Linux setups at this time.

kangas commented on docs/setup.md:
```diff
@@ -21,24 +21,19 @@ ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/
-sudo easy_install pip;
-pip install pyobjc;
-npm install -g node-gyp;
+brew install nvm;
```
Peter, all of your proposed changes in this README look terrific to me! Except for this one line.

`nvm` discourages installation via homebrew. Per https://github.com/creationix/nvm

> Homebrew installation is not supported. If you have issues with homebrew-installed nvm, please brew uninstall it, and install it using the instructions below, before filing an issue.

For our docs, I recommend that we point people to https://github.com/creationix/nvm#install-script and say "follow these instructions".

I generally hate recommending "curl -o <url> | bash" to install anything (which is what nvm actually recommends), but at least it's not "sudo bash".

pzrq replied:
@kangas Good find, I have updated the PR accordingly and also added a similar change for http://brew.sh/ immediately above, thanks.

pzrq committed:
Discourage brew installation of nvm as discovered by @kangas

pzrq committed:
Similarly to nvm, http://brew.sh/ has different instructions

So we should just point developers in the right direction.

i.e.
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
vs
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"

kangas reviewed (approved):
LGTM
